### PR TITLE
Integration Test CICD: iam role bugfix

### PR DESCRIPTION
**Pull request by SofiaSazonova**
### Feature or Bugfix

- Bugfix

### Detail
- remove unused trust policy that cause problems
- remove iam role after test

### Relates
- <URL or Ticket>

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests_new/integration_tests/aws_clients/iam.py [hunk 1]
```diff
@@ -21,13 +21,12 @@ def get_role(self, role_name):
             log.info(f'Error occurred: {e}')
```
Comment (noah-paige) [inline tests_new/integration_tests/aws_clients/iam.py:21]: nit: is `def create_role_if_not_exists(...)` method used or only `def get_consumption_role(...)` to get or create the iam role

If not used lets remove

**Review by noah-paige — COMMENTED** [review]
looks good - some minor clean up comments

File: tests_new/integration_tests/modules/share_base/test_new_crossacc_s3_share.py [hunk 2]
```diff
@@ -277,3 +277,7 @@ def test_revoke_succeeded(client1, share_params_main):
     assert_that(updated_share.status).is_equal_to('Processed')
     assert_that(items).extracting('status').contains_only('Revoke_Succeeded')
     assert_that(items).extracting('itemType').contains(*ALL_S3_SHARABLE_TYPES_NAMES)
+
+
+def test_1(consumption_role_1):
+    print(consumption_role_1.IAMRoleArn)
```
[the remark this answers is not in the document]
Comment (SofiaSazonova) [reply tests_new/integration_tests/modules/share_base/test_new_crossacc_s3_share.py:283]: Oops, forgot about it

**Review by noah-paige — APPROVED** [review]
looks gppd

**Review by petrkalos — APPROVED** [review]

**Pull request merged**